pkg/karmadactl/cmdinit/utils/kubeconfig_test.go

```diff
@@ -0,0 +1,154 @@
+					"local": {
+						Server:                   "https://127.0.0.1:6443",
+						CertificateAuthorityData: []byte{'c', 'a', 'c', 'e', 'r', 't'},
+					},
+				},
+				Contexts: map[string]*clientcmdapi.Context{
+					"local": {
+						Cluster:  "local",
+						AuthInfo: "admin",
+					},
+				},
+				AuthInfos:      map[string]*clientcmdapi.AuthInfo{},
+				CurrentContext: "local",
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if got := CreateBasic(tt.args.serverURL, tt.args.userName, tt.args.clusterName, tt.args.caCert); !reflect.DeepEqual(got, tt.want) {
+				t.Errorf("CreateBasic() = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func TestCreateWithCerts(t *testing.T) {
+	type args struct {
+		serverURL   string
+		userName    string
+		clusterName string
+		caCert      []byte
+		clientKey   []byte
+		clientCert  []byte
+	}
+	tests := []struct {
+		name string
+		args args
+		want *clientcmdapi.Config
+	}{
+		{
+			name: "create config from args with certs",
+			args: args{
+				serverURL:   "https://127.0.0.1:6443",
+				userName:    "admin",
+				clusterName: "local",
+				caCert:      []byte{'c', 'a', 'c', 'e', 'r', 't'},
+				clientKey:   []byte{'c', 'l', 'i', 'e', 'n', 't', 'k', 'e', 'y'},
+				clientCert:  []byte{'c', 'l', 'i', 'e', 'n', 't', 'c', 'e', 'r', 't'},
+			},
+			want: &clientcmdapi.Config{
+				Clusters: map[string]*clientcmdapi.Cluster{
+					"local": {
+						Server:                   "https://127.0.0.1:6443",
+						CertificateAuthorityData: []byte{'c', 'a', 'c', 'e', 'r', 't'},
+					},
+				},
+				Contexts: map[string]*clientcmdapi.Context{
+					"local": {
+						Cluster:  "local",
+						AuthInfo: "admin",
+					},
+				},
+				AuthInfos: map[string]*clientcmdapi.AuthInfo{
+					"admin": {
+						ClientKeyData:         []byte{'c', 'l', 'i', 'e', 'n', 't', 'k', 'e', 'y'},
+						ClientCertificateData: []byte{'c', 'l', 'i', 'e', 'n', 't', 'c', 'e', 'r', 't'},
+					},
+				},
+				CurrentContext: "local",
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if got := CreateWithCerts(tt.args.serverURL, tt.args.userName, tt.args.clusterName, tt.args.caCert, tt.args.clientKey, tt.args.clientCert); !reflect.DeepEqual(got, tt.want) {
+				t.Errorf("CreateWithCerts() = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func TestWriteKubeConfigFromSpec(t *testing.T) {
+	type args struct {
+		serverURL      string
+		userName       string
+		clusterName    string
+		kubeconfigPath string
+		kubeconfigName string
+		caCert         []byte
+		clientKey      []byte
+		clientCert     []byte
+	}
+	tests := []struct {
+		name    string
+		args    args
+		wantErr bool
+	}{
+		{
+			name: "Test Write KubeConfig From Spec",
+			args: args{
+				serverURL:      "https://127.0.0.1:6443",
+				userName:       "admin",
+				clusterName:    "local",
+				kubeconfigPath: "/tmp",
+				kubeconfigName: "karmada-test.kubeconfig",
```

Comment:
Hi @my-git9, this test case cannot pass the test on the Windows operating system:
```log
kubeconfig_test.go:148: WriteKubeConfigFromSpec() error = open \tmp\karmada-test.kubeconfig: The system cannot find the path specified., wantErr false
```

Reply:
nice finding. 
Please send another PR for this.

Reply:
Sorry, could you give me some advices for the path?

Reply:
Is it OK to generate a temporary file in the current directory?

Reply:
OK, I will try it later